lib/appsignal/rack/rails_instrumentation.rb

```diff
@@ -0,0 +1,41 @@
+require 'rack'
+
+module Appsignal
+  module Rack
+    class RailsInstrumentation
+      def initialize(app, options = {})
+        Appsignal.logger.debug 'Initializing Appsignal::Rack::RailsInstrumentation'
+        @app, @options = app, options
+      end
+
+      def call(env)
+        if Appsignal.active?
+          call_with_appsignal_monitoring(env)
+        else
+          @app.call(env)
+        end
+      end
+
+      def call_with_appsignal_monitoring(env)
+        request = ActionDispatch::Request.new(env)
+        transaction = Appsignal::Transaction.create(
+          env['action_dispatch.request_id'],
+          Appsignal::Transaction::HTTP_REQUEST,
+          request
+        )
+        begin
+          @app.call(env)
+        rescue => error
+          transaction.set_error(error)
+          raise error
+        ensure
+          transaction.set_http_or_background_action
+          transaction.set_http_or_background_queue_start
+          transaction.set_metadata('path', request.path)
```

Comment:
Is this how metadata works? I thought the key was the "group" and the value a hash of key/values. 

So `transaction.set_metadata('session', request.session)` would generate the session block on an exception view.